### ISSUE TITLE
Discussion about code style conventions

### ISSUE THREAD

**Issue by gilberto-torrezan** [GwtMaterialDesign/gwt-material#154]
Recently I opened a pull request that was rejected *a priori* due to code style problems. The main issue was the use of tabs instead of spaces (see #153).

I don't want to discuss which one is better (it's like a religion), but I do want to discuss what will be the general code style conventions for this project, to ease the process of contributing and to avoid surprises.

My suggestion when I reformatted the code was to use the [GWT code style](http://www.gwtproject.org/makinggwtbetter.html#codestyle) conventions. I even put that notice on the README file on the same pull request (better than nothing). But I do agree that some discussion should be made on the subject before really adopting (or creating) a convention.

Opinions?

**Comment by remarkNS** [comment]
I agree with you. GWT code style is good way to go. Please don't be angry on Ben I am sure he did not mean to offend you. This is the a good time go out in some bar and drink beer together, but unfortunately as I can tell, we are all from different continents... Please try to relax a bit and let's be a virtual friends again. Cheers !

**Comment by dlaunen** [comment]
+1 GWT code style conventions only make sense to adhere to since this is a
GWT project.


On Fri, Nov 20, 2015 at 8:04 AM, Gilberto Torrezan <notifications@github.com

**Comment by gilberto-torrezan** [comment]
I wasn't angry (and I'm sorry if it looked that way), but a bit surprised. But in the end it was a good issue, since it raised questions that can improve the project overall. =)

**Comment by BenDol** [comment]
Yeah, very surprised at the unreasonable response made on https://github.com/GwtMaterialDesign/gwt-material/pull/153, lets keep things civil. I was only adhering to a code standard that I use in my enterprise development work, since this project didn't have a conventional code style format before I reworked all the code. We don't need to adhere to a GWT code style just one that the team members all agree on. Personally I use 4 spaces rather than tabs (simple format setting change) simply because spaces are a consistent size on all different display parsers. Wasn't suppose to be an issue just a simple request.

Personally I don't think we need to completely reformat all the code when the current one is working fine as it is.

**Comment by remarkNS** [comment]
4 spaces and no tabs is "GWT code style" convention, does everybody agree to use this ?
I also do not think it is that important to reformat back all existing code, but I think it is important to follow some convention that is the conclusion of this discussion.

**Comment by gilberto-torrezan** [comment]
From the IDE standpoint, using a complete code convention or "just 4 spaces instead of tabs" is quite the same: in Eclipse, just tap Ctrl+Shift+F and everything is formatted.

Since its just a matter of pressing some keys (or not at all, if you bind the formatter to a save action), ensuring a full code convention could be preferable. Here is my point: I use style A, and someone else style B, and I reformat the code to A and push it. The second person would pull it, reformat to style B, and push again. Tracking the real code modifications would be a mess: git would show all reformatted lines instead of just what was really changed.

(Note, that just happened on [this commit](https://github.com/GwtMaterialDesign/gwt-material/commit/5c90c4f434b1cada9b529248f3f9dd8cc4aec236))

Anyway, I'm not GWT code style evangelist, I just think the code should look the same all over the project, regardless the author. We could use BenDol style template, or GWT template, or IBM Java template, whatever... as long as we agree on using one.

In other words: when contributing code, I'd love to discuss code structure, design decisions, performance, documentation, tests and so on, but *not* code style.

**Comment by BenDol** [comment]
This can be reopened if required in the future.

**Comment by kevzlou7979** [comment]
Ok i agree
